test/eth/next_interval_calculator.spec.js

```diff
@@ -1,189 +1,173 @@
-        assert.deepStrictEqual(worker.sleepTimeMsec, 0);
-    });
-
-    it('Node not called if Node is ahead', async function () {
-        const worker = new eth_worker.worker(constants);
-        worker.web3Wrapper = new MockIsCalledWeb3Wrapper();
-
-        worker.lastExportedBlock = 1;
-        worker.lastConfirmedBlock = 2;
-
-        await nextIntervalCalculator(worker);
-        assert.deepStrictEqual(worker.web3Wrapper.isCalled, false);
-    });
-
-    it('Block interval is not exceeded', async function () {
-        const worker = new eth_worker.worker(constants);
-
-        worker.lastExportedBlock = 1;
-        worker.lastConfirmedBlock = constants.BLOCK_INTERVAL + constants.CONFIRMATIONS + 10;
-
-        const result = await nextIntervalCalculator(worker);
-        assert.deepStrictEqual(result.success, true);
-        assert.deepStrictEqual(result.fromBlock, 2);
-        assert.deepStrictEqual(result.toBlock, 1 + constants.BLOCK_INTERVAL);
-    });
+  it('sets sleep time accordingly when we\'ve caught up', () => {
+    const worker = new eth_worker.worker(constants);
+    setWorkerSleepTime(worker, 2);
 
+    assert.deepStrictEqual(worker.sleepTimeMsec, worker.settings.LOOP_INTERVAL_CURRENT_MODE_SEC * 1000);
+  });
 });
 
-describe('Check logic when Node is not ahead', function () {
-    it('Check interval is correct if Node is behind', async function () {
-        const worker = new eth_worker.worker(constants);
-        worker.web3Wrapper = new MockWeb3Wrapper(constants.CONFIRMATIONS + 10);
-
-        worker.lastExportedBlock = 2;
-        worker.lastConfirmedBlock = 1;
-
-        const resultSecond = await nextIntervalCalculator(worker);
-        assert.deepStrictEqual(resultSecond.success, true);
-        assert.deepStrictEqual(resultSecond.fromBlock, 3);
-        assert.deepStrictEqual(resultSecond.toBlock, 10);
-    });
-
-    it('Sleep time set if Node is not ahead', async function () {
-        const worker = new eth_worker.worker(constants);
-        worker.web3Wrapper = new MockWeb3Wrapper(constants.CONFIRMATIONS + 10);
-
-        worker.lastExportedBlock = 1;
-        worker.lastConfirmedBlock = 1;
-
-        await nextIntervalCalculator(worker);
-        assert.deepStrictEqual(worker.sleepTimeMsec, constants.LOOP_INTERVAL_CURRENT_MODE_SEC * 1000);
-    });
-
-    it('Node gets called if Node is not ahead', async function () {
-        const worker = new eth_worker.worker(constants);
-        worker.web3Wrapper = new MockIsCalledWeb3Wrapper();
+describe('nextIntervalCalculator', () => {
+  it('returns interval according to the worker progress', () => {
+    const worker = new eth_worker.worker(constants);
+    worker.lastExportedBlock = 0;
+    worker.lastConfirmedBlock = 150;
+
+    const { fromBlock, toBlock } = nextIntervalCalculator(worker);
+    assert.deepStrictEqual(fromBlock, 1);
+    assert.deepStrictEqual(toBlock, 100);
+  });
+
+  it('returns the interval accordingly when lastConfirmedBlock is close', () => {
```

Comment:
Is this the previously called: 'Exporter should not wait for full BLOCK_INTERVAL' test? Don't you find the previous more describing of exactly what is happening?

Reply:
I'm not sure, the waiting bit implies something else IMO. But I'll make it more clear, I don't know why I've named them in such a mysterious fashion